--- src/include/wlan/wlan_compat.h.orig	Thu Aug 28 16:00:14 2003
+++ src/include/wlan/wlan_compat.h	Tue Sep 30 21:56:06 2003
@@ -270,11 +270,12 @@
 #define WLAN_DBVAR	wlan_debug
 #endif
 
-#define WLAN_LOG_ERROR(x,args...) printk(KERN_ERR "%s: " x , __FUNCTION__ , ##args);
 
-#define WLAN_LOG_WARNING(x,args...) printk(KERN_WARNING "%s: " x , __FUNCTION__, ##args);
+#define WLAN_LOG_ERROR(x,args...) printk(KERN_ERR  x , ##args);
 
-#define WLAN_LOG_NOTICE(x, args...) printk(KERN_NOTICE "%s: " x , __FUNCTION__, ##args);
+#define WLAN_LOG_WARNING(x,args...) printk(KERN_WARNING  x , ##args);
+
+#define WLAN_LOG_NOTICE(x, args...) printk(KERN_NOTICE x , ##args);
 
 #define WLAN_LOG_INFO(args... ) printk(KERN_INFO args)
 
@@ -290,7 +291,7 @@
 	#define DBFENTER { if ( WLAN_DBVAR >= 5 ){ WLAN_LOG_DEBUG(3,"---->\n"); } }
 	#define DBFEXIT  { if ( WLAN_DBVAR >= 5 ){ WLAN_LOG_DEBUG(3,"<----\n"); } }
 
-	#define WLAN_LOG_DEBUG(l,x,args...) if ( WLAN_DBVAR >= (l)) printk(KERN_DEBUG "%s: " x ,  __FUNCTION__ , ##args );
+	#define WLAN_LOG_DEBUG(l,x,args...) if ( WLAN_DBVAR >= (l)) printk(KERN_DEBUG x , ##args );
 #else
 	#define WLAN_ASSERT(c) 
 	#define WLAN_HEX_DUMP( l, s, p, n)
